### PR TITLE
[digital_guardian] Correct time formats, data size parsing

### PR DESCRIPTION
**Pull request by chrisberkhout**
## Proposed commit message

```
[digital_guardian] Correct time formats, data size parsing

- Correct the `dg_time` format (may have AM/PM).
- Correct the `dg_processed_time` format (to `epoch_millis`).
- Process additional timestamp fields: `dg_local_timestamp`, `pi_fal`,
  `pi_fcl`, `pi_fml`.
- Add fallback/alternate formats for the string-valued time fields.
- For data size fields, add parsing logic.
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)

### CODE REVIEW

File: packages/digital_guardian/data_stream/arc/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -89,28 +127,70 @@ processors:
         def sev = ctx.digital_guardian.arc.inc_sev;
         ctx.event.severity = params.getOrDefault(sev, params['Unknown']);
 
-# Date fields
+# Convert date fields
```
Comment (efd6) [inline packages/digital_guardian/data_stream/arc/elasticsearch/ingest_pipeline/default.yml:130]: Do we know what timezones these civil timestamps are in?

Comment (chrisberkhout) [reply packages/digital_guardian/data_stream/arc/elasticsearch/ingest_pipeline/default.yml:130]: No. The few examples of full data I have don't contain any information about time zone or location or IP address. Digital Guardian ARC is a cloud product that runs in AWS. I suspect `dg_time` is the server time in UTC and `dg_local_timestamp` is the local time on an endpoint or data source. It looks like there are multiple time zones in the data:

image

Comment (efd6) [reply packages/digital_guardian/data_stream/arc/elasticsearch/ingest_pipeline/default.yml:130]: How do we want to deal with that?

Comment (efd6) [reply packages/digital_guardian/data_stream/arc/elasticsearch/ingest_pipeline/default.yml:130]: Resolved off-line; we just treat these as UTC.

**Review by efd6 — APPROVED** [review]
LGTM after conflicts resolved.

**Pull request merged**